src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -0,0 +1,1781 @@
+            db_session=db_session,
+            action_id=IncidentReportActions.project_select,
+            dispatch_action=True,
+        ),
+    ]
+
+    modal = Modal(
+        title="Report Incident",
+        blocks=blocks,
+        submit="Submit",
+        close="Cancel",
+        callback_id=IncidentReportActions.submit,
+        private_metadata=context["subject"].json(),
+    ).build()
+
+    await client.views_open(trigger_id=body["trigger_id"], view=modal)
+
+
+async def ack_report_incident_submission_event(ack):
+    """Handles the report incident submission event acknowledgment."""
+    modal = Modal(
+        title="Report Incident",
+        close="Close",
+        blocks=[Section(text="Creating incident resources...")],
+    ).build()
+    await ack(response_action="update", view=modal)
+
+
+async def handle_report_incident_submission_event(user, client, body, form_data):
+    """Handles the report incident submission"""
+    # refetch session as we can't pass a db_session lazily, these could be moved to @background_task functions
+    # in the future
+
+    # TODO handle multiple organizations during submission
+    schema_engine = engine.execution_options(
+        schema_translate_map={
+            None: "dispatch_organization_default",
+        }
+    )
+    db_session = sessionmaker(bind=schema_engine)()
+
+    tags = []
+    for t in form_data.get(DefaultBlockIds.tags_multi_select, []):
+        # we have to fetch as only the IDs are embedded in slack
+        tag = tag_service.get(db_session=db_session, tag_id=int(t["value"]))
+        tags.append(tag)
+
+    project = {"name": form_data[DefaultBlockIds.project_select]["name"]}
+
+    incident_type = None
+    if form_data.get(DefaultBlockIds.incident_type_select):
+        incident_type = {"name": form_data[DefaultBlockIds.incident_type_select]["name"]}
+
+    incident_priority = None
+    if form_data.get(DefaultBlockIds.incident_priority_select):
+        incident_priority = {"name": form_data[DefaultBlockIds.incident_priority_select]["name"]}
+
+    incident_severity = None
+    if form_data.get(DefaultBlockIds.incident_severity_select):
+        incident_severity = {"name": form_data[DefaultBlockIds.incident_severity_select]["name"]}
+
+    incident_in = IncidentCreate(
+        title=form_data[DefaultBlockIds.title_input],
+        description=form_data[DefaultBlockIds.description_input],
+        incident_type=incident_type,
+        incident_priority=incident_priority,
+        incident_severity=incident_severity,
+        project=project,
+        reporter=ParticipantUpdate(individual=IndividualContactRead(email=user.email)),
+        tags=tags,
+    )
+
+    blocks = [
+        Section(text="Creating your incident..."),
+    ]
+
+    modal = Modal(title="Incident Report", blocks=blocks, close="Close").build()
+
+    result = await client.views_update(
+        view_id=body["view"]["id"],
+        trigger_id=body["trigger_id"],
+        view=modal,
+    )
```

Comment:
passing in the hash from `body` here resulted in a hash conflict and prevented the incident from being created. The hash we got here in `body` was associated with the first `Report Incident` view and I suspect we want the hash of the `Creating incident resources...` view above. It's unclear to me how to send this hash along to this function. I found what looks like a similar issue here: https://github.com/slackapi/bolt-js/issues/787

Nonetheless, removing the hash check resolved the issue, and this function appears to be the odd one out in terms of validating the hash of a view that leverages an `ack=` handler/listener.